elytron/ELY-1712-enhanced-audit-logging.adoc

```diff
@@ -0,0 +1,283 @@
+** Specifying the criteria, the value(s), and the configuration changes should be done as follows:
+*** When the programmatic approach is used, a method call should be available to the endpoint which will accept a given criteria, the corresponding values, and the filtering options
+*** When the markup approach is used, the inputs should be specified via attributes ```criteria="supported_criteria"```, ```values="respective_value_input"```, ```filtering="option1=change,option2=change,..."```.
+When this approach is used, these attributes should be available as attributes under the ```auditLogType``` in the Elytron schema
+*** Further information on the criteria and values can be found in the first table below
+** Example for specifying a criteria which has string values:
+[source,java]
+----
+String criteria = "type";
+List<String> typeValues = new ArrayList<>(Arrays.asList("authentication", "failure"));
+Map<String, String> filtering = new HashMap<>();
+filtering.put("location", "/home/audit/logs");
+filtering.put("suffix", ".yyyy-MM-dd_HHmm");
+AuditEndpoint endpoint = PeriodicRotatingFileAuditEndpoint.builder()
+        .setTimeZone(UTC)
+        .setSuffix(".yyyy-MM-ww-dd-a-HH_mm")
+        .setLocation(logFile)
+        .setClock(clock)
+        .setFiltering(criteria, typeValues, filtering)
+        .build();
+----
+[source,xml]
+----
+... criteria="type" values="authentication,failure" filtering="location='/home/audit/logs',suffix='.yyyy-MM-dd_HHmm'" ...
+----
+** Example for specifying a criteria which has int values
+[source,java]
+----
+String criteria = "facility";
+int lowRange = 0;
+int highRange = 15;
+Map<String, String> filtering = new HashMap<>();
+filtering.put("location", "/home/audit/logs");
+filtering.put("suffix", ".yyyy-MM-dd_HHmm");
+AuditEndpoint endpoint = PeriodicRotatingFileAuditEndpoint.builder()
+        .setTimeZone(UTC)
+        .setSuffix(".yyyy-MM-ww-dd-a-HH_mm")
+        .setLocation(logFile)
+        .setClock(clock)
+        .setFiltering(criteria, lowRange, highRange, filtering)
+        .build();
+----
+[source,xml]
+----
+... criteria="facility" values="0,15" filtering="location='/home/audit/logs',suffix='.yyyy-MM-dd_HHmm'" ...
+----
+
+* Logging should be done by using the proposal API as described in https://github.com/wildfly/wildfly-proposals/pull/181[WFCORE-4355]
+** A StdoutAuditEndpoint should be added to WildFly-Core that uses this API to log an event to Stdout
+** An AysncAuditEndpoint should be added to WildFly-Core that uses this API to log events to a queue before they're writing to Stdout
+** JsonSecurityEventFormatter should be moved to WildFly-Core and modified to use the JsonFormatter in this API
+** The current FileAuditEndpoints, FileAuditEndpoint, PeriodicRotatingFileAuditEndpoint, and SizeRotatingFileAuditEndpoint
+should be moved to WildFly-Core
```

Comment:
What changes will this introduce to model (if any)?

Reply:
I don't think this would introduce any changes to users other than additional Endpoints being available

Reply:
Thanks! Where/How can user access them?

Reply:
These would be additional attributes in the model, I'll add in their names to the proposal